File: cp_rust/src/bin/abc042-b-iroha-loves-strings-abc-edition.rs
// Created by Ayush Biswas at 2026/01/20 14:13
// https://atcoder.jp/contests/abc042/tasks/abc042_b
use cp_lib::*;

// @code begin
use cpio::*;
use itertools::Itertools;

sol! {
    fn solution(
        [n, _l] : [usize; 2],
        strings : [String]; n
    ) -> String {
        strings.into_iter().sorted().collect_vec().join("")
    }
}
// @code end

File: cp_haskell/bin/abc042-b-iroha-loves-strings-abc-edition.hs
-- Created by Ayush Biswas at 2026/03/01 11:53
-- https://atcoder.jp/contests/abc042/tasks/abc042_b

-- @code begin

import Control.Monad (replicateM)
import Data.List

main :: IO ()
main = do
  [n, l] :: [Int] <- map read . words <$> getLine
  x <- concat . sort <$> replicateM n getLine
  putStrLn x

-- @code end